Library/OcBootManagementLib/VariableManagement.c

```diff
@@ -312,6 +314,109 @@ InternalGetBootstrapBootData (
   return OptionData;
 }
 
+EFI_STATUS
+OcGetSip (
+  UINT32 *CsrActiveConfig,
+  UINT32 *Attributes
+  )
+{
+  EFI_STATUS      Status;
+  UINTN           DataSize;
+
+  *CsrActiveConfig = 0;
+  DataSize = sizeof (*CsrActiveConfig);
+
+  Status = gRT->GetVariable (
+    L"csr-active-config",
+    &gAppleBootVariableGuid,
+    Attributes,
+    &DataSize,
+    CsrActiveConfig);
+
+  return Status;
+}
+
+EFI_STATUS
+OcSetSip (
+  UINT32 *CsrActiveConfig,
+  UINT32 Attributes
+  )
+{
+  EFI_STATUS      Status;
+
+  Status = gRT->SetVariable (
+    L"csr-active-config",
+    &gAppleBootVariableGuid,
+    Attributes,
+    CsrActiveConfig == NULL ? 0 : sizeof (*CsrActiveConfig),
+    CsrActiveConfig);
+
+  return Status;
+}
+
+BOOLEAN
+OcIsSipEnabled (
+  EFI_STATUS  FetchStatus,
+  UINT32      CsrActiveConfig
+  )
+{
+  return FetchStatus == EFI_NOT_FOUND || (CsrActiveConfig & ~CSR_ALLOW_APPLE_INTERNAL) == 0;
+}
+
+EFI_STATUS
+OcToggleSip (
+  UINT32 CsrActiveConfig
+  )
+{
+  EFI_STATUS  Status;
+  UINT32      Attributes;
+  UINT32      CsrConfig;
+
+  //
+  // Use existing attributes where present (e.g. keep changes made while WriteFlash = false as volatile only)
+  //
+  Status = OcGetSip (&CsrConfig, &Attributes);
+
+  if (Status != EFI_NOT_FOUND && EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "OCB: Error getting SIP status - %r\n", Status));
+  } else {
+      if (Status == EFI_NOT_FOUND) {
+        Attributes = CSR_APPLE_SIP_NVRAM_NV_ATTR;
+      } else {
+        Attributes &= CSR_APPLE_SIP_NVRAM_NV_ATTR;
+      }
+
+    if (OcIsSipEnabled (Status, CsrConfig)) {
+      CsrConfig = CsrActiveConfig;
+      Status = OcSetSip(&CsrConfig, Attributes);
+      if (EFI_ERROR (Status)) {
+        DEBUG ((DEBUG_ERROR, "OCB: Error disabling SIP - r\n", Status));
+      }
+    } else {
+      CsrConfig = CSR_ALLOW_APPLE_INTERNAL;
```

Comment:
Enabled SIP is just 0. This makes no sense as we are not targeting Apple Internal builds here.

Reply:
It's not just 0.
This is matching the behaviour of my Mac.
Oh.
It's because I'm on a beta version. Possibly.
But if it is we still need to support that.
There is definitely not nothing going on, here!

Reply:
We do not need to support this. Consumer builds should not have AppleInternal, this is a bug of your beta at most.

Reply:
This is weird, It's running like this on different versions of the Big Sur beta, on my MBP and on my Dell hack.
I could leave in support for showing that when just this bit is set in csr-active-config, SIP is still considered enabled (in fact, I probably should, since that is definitely what csrutil does - you can dbl check this, on your setup, if you have a moment?), but then still not set it as our default for enabled and rather use 0 as it should be.

Reply:
OK. You can show, but you really must not set this bit at all.

Reply:
Sure, will change.